exercise/exercise_models.py

```diff
@@ -275,6 +275,32 @@ def get_templates(self):
         entries = pick_localized(self.templates, get_language())
         return [(url,url.split('/')[-1]) for url in entries.split()]
 
+    def intersect_form_spec_keys(self, another_lobject, include_static_fields=False):
+        """Return the intersection of form field keys from this learning object and another.
+        In other words, keys that exist in both forms. The form keys are read
+        from the form_spec list of the exercise_info JSON field.
+        """
+        if not (self.exercise_info and another_lobject.exercise_info \
+                and 'form_spec' in self.exercise_info \
+                and 'form_spec' in another_lobject.exercise_info):
+            return set()
```

Comment:
Sorry, missed this line. Yeah. This handles 90% of the problems. Only thing to check more, is to ensure exercise_infos are dicts.